src/strands_tools/elasticsearch_memory.py

```diff
@@ -0,0 +1,675 @@
+            memories.append(memory)
+
+        result = {
+            "memories": memories,
+            "total": response["hits"]["total"]["value"],
+            "max_score": response["hits"]["max_score"],
+            "search_info": {
+                "query_embedding_generated": True,
+                "search_type": "k-NN vector similarity",
+                "embedding_model": self.embedding_model,
+                "embedding_dimensions": DEFAULT_EMBEDDING_DIMS,
+                "similarity_function": "cosine",
+            },
+        }
+
+        # Add next_token if there are more results
+        if from_offset + max_results < response["hits"]["total"]["value"]:
+            result["next_token"] = str(from_offset + max_results)
+
+        return result
+
+    def _list_memories(self, max_results: int, next_token: Optional[str] = None) -> Dict:
+        """
+        List all memories in the namespace.
+
+        Args:
+            max_results: Maximum number of results
+            next_token: Pagination token
+
+        Returns:
+            Dict containing all memories
+        """
+        # Calculate offset from next_token
+        from_offset = int(next_token) if next_token else 0
+
+        search_body = {
+            "query": {"term": {"namespace": self.namespace}},
+            "sort": [{"timestamp": {"order": "desc"}}],
+            "from": from_offset,
+            "size": max_results,
+            "_source": ["memory_id", "content", "timestamp", "metadata"],
+        }
+
+        response = self.es_client.search(index=self.index_name, body=search_body)
+
+        # Format results
+        memories = []
+        for hit in response["hits"]["hits"]:
+            memory = {
+                "memory_id": hit["_source"]["memory_id"],
+                "content": hit["_source"]["content"],
+                "timestamp": hit["_source"]["timestamp"],
+                "metadata": hit["_source"].get("metadata", {}),
+            }
+            memories.append(memory)
+
+        result = {"memories": memories, "total": response["hits"]["total"]["value"]}
+
+        # Add next_token if there are more results
+        if from_offset + max_results < response["hits"]["total"]["value"]:
+            result["next_token"] = str(from_offset + max_results)
+
+        return result
+
+    def _get_memory(self, memory_id: str) -> Dict:
+        """
+        Get a specific memory by ID.
+
+        Args:
+            memory_id: Memory ID to retrieve
+
+        Returns:
+            Dict containing the memory
+
+        Raises:
+            Exception: If memory not found or not in correct namespace
+        """
+        try:
+            response = self.es_client.get(index=self.index_name, id=memory_id)
+            source = response["_source"]
+
+            # Verify namespace
+            if source.get("namespace") != self.namespace:
+                raise Exception(f"Memory {memory_id} not found in namespace {self.namespace}")
+
+            return {
```

Comment:
Can we use { "status": "error" | "success", "content": [ { "text": "…" }, { "json": … } way of returning the results from decorated tools?